ugs-platform/ugs-platform-plugin-pendant/src/main/java/com/willwinder/ugs/platform/plugin/pendant/PendantTopComponent.java

```diff
@@ -0,0 +1,138 @@
+/**
+ * An interface to start and stop the pendant server.
+ */
+
+/*
+    Copywrite 2016 Will Winder
+
+    This file is part of Universal Gcode Sender (UGS).
+
+    UGS is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    UGS is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with UGS.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.willwinder.ugs.platform.plugin.pendant;
+
+import javax.swing.ImageIcon;
+import javax.swing.JLabel;
+import net.miginfocom.swing.MigLayout;
+import org.netbeans.api.settings.ConvertAsProperties;
+import org.openide.awt.ActionID;
+import org.openide.awt.ActionReference;
+import org.openide.windows.TopComponent;
+import org.openide.util.NbBundle.Messages;
+
+/**
+ * Top component which displays something.
+ */
+@ConvertAsProperties(
+        dtd = "-//com.willwinder.ugs.platform.plugin.pendant//Pendant//EN",
+        autostore = false
+)
+@TopComponent.Description(
+        preferredID = "PendantTopComponent",
+        //iconBase="SET/PATH/TO/ICON/HERE", 
+        persistenceType = TopComponent.PERSISTENCE_ALWAYS
+)
+@TopComponent.Registration(mode = "output", openAtStartup = false)
+@ActionID(category = "Window", id = "com.willwinder.ugs.platform.plugin.pendant.PendantTopComponent")
+@ActionReference(path = "Menu/Window" /*, position = 333 */)
+@TopComponent.OpenActionRegistration(
+        displayName = "#CTL_PendantAction",
+        preferredID = "PendantTopComponent"
+)
+@Messages({
+    "CTL_PendantAction=Pendant",
+    "CTL_PendantTopComponent=Pendant Window",
+    "HINT_PendantTopComponent=This is a Pendant window"
+})
+public final class PendantTopComponent extends TopComponent {
```

Comment:
There's probably no need to put this in a `TopComponent`. Take a look at `RunActionService` for an example of how to create a Netbeans `ServiceProvider`.

Services aren't initialized until they're used, so you would pair the `PendantServer` would need to be initialized with a `Lookup.getDefault().lookup(PendantServer.class);` call in an action like `PendantAction.java`.